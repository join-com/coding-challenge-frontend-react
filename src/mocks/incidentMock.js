const incidentMock = {
  id: 24134,
  title: "Stolen di Florino (red and white)",
  description: "",
  address: "Berlin, 10245, DE",
  occurred_at: 1402984800,
  updated_at: 1545551568,
  url: "https://bikewise.org/api/v1/incidents/24134",
  source: {
    name: "BikeIndex.org",
    html_url: "https://bikeindex.org/bikes/27159",
    api_url: "https://bikeindex.org/api/v1/bikes/27159"
  },
  media: {
    image_url: "https://bikeindex.s3.amazonaws.com/uploads/Pu/9564/large_dsc_1293.jpg",
    image_url_thumb: "https://bikeindex.s3.amazonaws.com/uploads/Pu/9564/small_dsc_1293.jpg"
  },
  location_type: null,
  location_description: null,
  type: "Theft",
  type_properties: null
};

export default incidentMock;
