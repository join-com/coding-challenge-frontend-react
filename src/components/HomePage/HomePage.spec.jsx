import React from "react";
import { shallow } from "enzyme";
import Component from "./HomePage";
import incidentMock from "../../mocks/incidentMock";

describe("HomePage component test", () => {
  const defaultProps = {
    incidents: [incidentMock],
    searchValue: "",
    itemsPerPage: 10,
    isLoading: false,
    error: null,
    totalIncidentsLength: 22,
    fetchIncidents: jest.fn(),
    changeUi: jest.fn()
  };

  const getWrapper = (testProps = {}) => {
    const props = { ...defaultProps, ...testProps };
    return shallow(<Component {...props} />);
  };

  it("should match snapshot and call fetchIncidents prop function", () => {
    const wrapper = getWrapper();
    expect(wrapper).toMatchSnapshot();
    expect(defaultProps.fetchIncidents).toHaveBeenCalledTimes(1);
  });

  it("should match snapshot for loading state", () => {
    const wrapper = getWrapper({ isLoading: true });
    expect(wrapper).toMatchSnapshot();
  });

  it("should match snapshot with error", () => {
    const wrapper = getWrapper({ error: "error message" });
    expect(wrapper).toMatchSnapshot();
  });

  it("should match snapshot empty results", () => {
    const wrapper = getWrapper({ totalIncidentsLength: 0, incidents: [] });
    expect(wrapper).toMatchSnapshot();
  });

  it("should correct handle changeSearchValue", () => {
    const changeUi = jest.fn();
    const event = { target: { value: 1 } };
    const wrapper = getWrapper({ changeUi });
    wrapper.instance().changeSearchValue(event);
    expect(changeUi).toBeCalledTimes(2);
    expect(changeUi).toHaveBeenNthCalledWith(1, { name: "searchValue", value: 1 });
    expect(changeUi).toHaveBeenNthCalledWith(2, { name: "currentPage", value: 0 });
  });

  it("should correct handle changeShowItemsPerPage", () => {
    const changeUi = jest.fn();
    const wrapper = getWrapper({ changeUi });
    const event = { target: { value: 10 } };
    wrapper.instance().changeShowItemsPerPage(event);
    expect(changeUi).toBeCalledTimes(2);
    expect(changeUi).toHaveBeenNthCalledWith(1, { name: "itemsPerPage", value: 10 });
    expect(changeUi).toHaveBeenNthCalledWith(2, { name: "currentPage", value: 0 });
  });
});
