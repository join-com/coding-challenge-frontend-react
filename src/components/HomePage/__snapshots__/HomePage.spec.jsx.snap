// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage component test should match snapshot and call fetchIncidents prop function 1`] = `
<styled.div>
  <Filter
    changeShowItemsPerPage={[Function]}
    disableInput={false}
    itemsPerPage={10}
    onChange={[Function]}
    totalIncidentsLength={22}
    value=""
  />
  <IncidentsList
    items={
      Array [
        Object {
          "address": "Berlin, 10245, DE",
          "description": "",
          "id": 24134,
          "location_description": null,
          "location_type": null,
          "media": Object {
            "image_url": "https://bikeindex.s3.amazonaws.com/uploads/Pu/9564/large_dsc_1293.jpg",
            "image_url_thumb": "https://bikeindex.s3.amazonaws.com/uploads/Pu/9564/small_dsc_1293.jpg",
          },
          "occurred_at": 1402984800,
          "source": Object {
            "api_url": "https://bikeindex.org/api/v1/bikes/27159",
            "html_url": "https://bikeindex.org/bikes/27159",
            "name": "BikeIndex.org",
          },
          "title": "Stolen di Florino (red and white)",
          "type": "Theft",
          "type_properties": null,
          "updated_at": 1545551568,
          "url": "https://bikewise.org/api/v1/incidents/24134",
        },
      ]
    }
  />
  <Connect(Pagination) />
</styled.div>
`;

exports[`HomePage component test should match snapshot empty results 1`] = `
<styled.div>
  <Filter
    changeShowItemsPerPage={[Function]}
    disableInput={false}
    itemsPerPage={10}
    onChange={[Function]}
    totalIncidentsLength={0}
    value=""
  />
  <Text
    bold={false}
    size="normal"
    uppercase={false}
  >
    No results
  </Text>
</styled.div>
`;

exports[`HomePage component test should match snapshot for loading state 1`] = `
<styled.div>
  <Filter
    changeShowItemsPerPage={[Function]}
    disableInput={true}
    itemsPerPage={10}
    onChange={[Function]}
    totalIncidentsLength={22}
    value=""
  />
  <Text
    bold={false}
    size="normal"
    uppercase={false}
  >
    Loading ...
  </Text>
</styled.div>
`;

exports[`HomePage component test should match snapshot with error 1`] = `
<styled.div>
  <Filter
    changeShowItemsPerPage={[Function]}
    disableInput={false}
    itemsPerPage={10}
    onChange={[Function]}
    totalIncidentsLength={22}
    value=""
  />
  <ErrorMessage
    message="error message"
  />
</styled.div>
`;
